Lib/cu2qu/rf.py

```diff
@@ -46,6 +44,103 @@ def zip(*args):
     return _zip(*args)
 
 
+class GetSegmentsPen(AbstractPen):
+    """Pen to collect segments into lists of points for conversion."""
+
+    def __init__(self):
+        self._last_pt = None
+        self.segments = []
+
+    def _add_segment(self, tag, *args):
+        self.segments.append((tag, args))
+
+    def moveTo(self, pt):
+        self._add_segment('move', pt)
+        self._last_pt = pt
+
+    def lineTo(self, pt):
+        self._add_segment('line', pt)
+        self._last_pt = pt
+
+    def curveTo(self, *points):
```

Comment:
I think that the first point on any contour is a moveTo, so this should always be ok.